### PR TITLE
Revamp implementation and add arithmetic operations support for byte type 

### PR DESCRIPTION
**Pull request by pranavan15**
## Purpose
This PR modifies the Ballerina byte type implementation. With this PR, `byte` type has been moved from int registry to long registry and treated as same as `int` type. This PR also adds arithmetic operations support for byte type. The existing bitwise operations have been modified and missing operations are added with this PR.

Arithmatic operations such as `+, -, *, /, %` are now supported between `byte, byte` and `byte, int`. 

This PR Fixes,
* https://github.com/ballerina-platform/ballerina-lang/issues/13821
* https://github.com/ballerina-platform/ballerina-lang/issues/13865

### CODE REVIEW

File: bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVMStructs.java [hunk 1]
```diff
@@ -104,9 +104,11 @@ private static BValue getBValue(BType type, Object value) {
                 break;
             case TypeTags.BYTE_TAG:
                 if (value instanceof Byte) {
-                    return new BByte(((Byte) value));
+                    return new BByte(((Byte) value).longValue());
                 } else if (value instanceof Integer) {
-                    return new BByte(((Integer) value).byteValue());
+                    return new BByte(((Integer) value).longValue());
```
Comment (hasithaa) [inline bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVMStructs.java]: @pranavan15 will this guarantee the byte overflow?

File: bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BVM.java [hunk 2]
```diff
@@ -1101,11 +1087,13 @@ private static Strand invokeCallable(Strand ctx, BFunctionPointer fp, FunctionCa
...
     private static int expandLongRegs(StackFrame sf, BFunctionPointer fp) {
         int longIndex = 0;
-        if (fp.getAdditionalIndexCount(BTypes.typeInt.getTag()) > 0) {
+        if (fp.getAdditionalIndexCount(BTypes.typeInt.getTag()) > 0 ||
+                fp.getAdditionalIndexCount(BTypes.typeByte.getTag()) > 0) {
             if (sf.longRegs == null) {
                 sf.longRegs = new long[0];
             }
-            long[] newLongRegs = new long[sf.longRegs.length + fp.getAdditionalIndexCount(BTypes.typeInt.getTag())];
+            long[] newLongRegs = new long[sf.longRegs.length + fp.getAdditionalIndexCount(BTypes.typeInt.getTag()) +
```
Comment (hasithaa) [inline bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BVM.java]: Can't we simply use int index here for byte as well? Basically, fix this at the code generator level.

File: bvm/ballerina-core/src/main/java/org/ballerinalang/channels/DatabaseUtils.java [hunk 3]
```diff
@@ -281,7 +281,7 @@ private static BValue getValue(ResultSet resultSet, BType bType) throws SQLExcep
             case TypeTags.STRING_TAG:
                 return new BString(resultSet.getString(2));
             case TypeTags.BYTE_TAG:
-                return new BByte(resultSet.getByte(2));
+                return new BByte(Byte.toUnsignedLong(resultSet.getByte(2)));
```
Comment (hasithaa) [inline bvm/ballerina-core/src/main/java/org/ballerinalang/channels/DatabaseUtils.java]: Seems like we have to use Byte.toUnsignedLong every time with BByte constructor. Can we simplify this?

File: bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVMStructs.java (hunk 1 above)
Comment (pranavan15) [reply bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVMStructs.java]: I have updated this logic with commit 51d1715 to handle possible byte overflow and underflow.

File: bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BVM.java (hunk 2 above)
Comment (pranavan15) [reply bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BVM.java]: Commit 4532e09 addresses this.

File: bvm/ballerina-core/src/main/java/org/ballerinalang/channels/DatabaseUtils.java (hunk 3 above)
Comment (pranavan15) [reply bvm/ballerina-core/src/main/java/org/ballerinalang/channels/DatabaseUtils.java]: Ack; Simplified via commit f3c8993 by adding a new BByte constructor that accepts a byte value.

**Pull request merged**